libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -559,6 +559,14 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			arguments.push_back(callArguments[std::distance(callArgumentNames.begin(), it)]);
 		}
 
+	if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
+		if (auto expressionType = dynamic_cast<TypeType const*>(memberAccess->expression().annotation().type))
+			if (auto contract = dynamic_cast<ContractDefinition const*>(expressionType->actualType()))
+				solUnimplementedAssert(
```

Comment:
Isn't this already caught when the function is eventually called?

Reply:
Depends. For example for external calls it will go into `appendExternalFunctionCall()` and maybe fail or maybe just do nothing or even do something wrong (haven't checked). I thought it would be a good practice to explicitly forbid cases that we know are not implemented. I sometimes run into cases where `solc --ir` fails or does nothing (e.g. modifiers inherited internal constructors from the previous PR) and I'm never really sure whether it's something that's not implemented or just broken. Hitting `solUnimplementedAssert` would make the situation much clearer to me.

Reply:
ok!